src/ZoneServer/Buffs/Handlers/Archer/Wugushi/Archer_VerminPot_Debuff.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Threading.Tasks;
+using Melia.Shared.Game.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.World.Actors;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Buffs.Handlers.Archer.Wugushi
+{
+	/// <summary>
+	/// Handle for the VerminPot Debuff (PoisonPot), which ticks damage every second.
+	/// </summary>
+	[BuffHandler(BuffId.Archer_VerminPot_Debuff)]
+	public class Archer_VerminPot_Debuff : BuffHandler
+	{
+		public override async void WhileActive(Buff buff)
+		{
+			if (buff.Target.IsDead)
+				return;
+
+			if (buff.Caster.TryGetSkill((SkillId)buff.NumArg2, out var skill))
+			{
+				var damageMultiplier = 1f;
+
+				if (buff.Caster.TryGetBuff(BuffId.Zhendu_Buff, out var ZhenduBuff))
+					damageMultiplier = ZhenduBuff.NumArg1;
+
+				var skillHitResult = SCR_SkillHit(buff.Caster, buff.Target, skill);
+				skillHitResult.Damage *= damageMultiplier;
+
+				// The damage amount is unknow, for now we are dealing
+				// the same amount as the original skill does
+				buff.Target.TakeDamage(skillHitResult.Damage, buff.Caster);
+
+				var hit = new HitInfo(buff.Caster, buff.Target, SkillId.Wugushi_ThrowGuPot, skillHitResult.Damage, HitResultType.Buff26);
+
+				Send.ZC_HIT_INFO(buff.Caster, buff.Target, hit);
+			}
+
+			var damageThickDelay = 1000f;
+			var skillLevel = (int)buff.NumArg1;
+
+			Crescendo_Bane_Buff.TryApply(buff.Caster, ref damageThickDelay);
+
+			await Task.Delay(TimeSpan.FromMilliseconds(damageThickDelay));
```

Comment:
it is the delay between the damages.
The Crescendo Bane Buff decreases the interval between each hit.
That's the reason.

Reply:
But you're not doing anything after the delay, are you? This isn't a loop, there's nothing coming after the `await`. You're creating the variable, you modify it, you create a task to wait until the delay is over, and then nothing happens. I don't know what you're trying to do there, but unfortunately I don't think you're accomplishing it^^"

Edit: After reading your other comment, it seems like you're trying to decrease the time until `WhileActive` is called again? Do I have that right? If so, I think you have the wrong idea about how buff updates work. The delay doesn't do anything, because there's nothing to do, and it runs on a task, independently of the buff updates. Even if there was something to do, it wouldn't affect the buff updates. What you would have to do is modify the time the buff uses to calculate the next update time, which it currently uses the buff's data for.

Unless I have it wrong, in which case you'll have to walk me through what exactly your intention is here.

Reply:
I have done this to fix it:

```
		public override void OnStart(Buff buff)
		{
			buff.Vars.SetInt(AdditionalHits, AdditionalHitsCount);

			var damageThickDelay = buff.Data.UpdateTime;

			Crescendo_Bane_Buff.TryApply(buff.Caster, ref damageThickDelay);

			buff.Data.UpdateTime = damageThickDelay;
		}
```

it seems to be working fine.

Reply:
That will only work as long as you're the only person on the server, because `Data` is a global reference. Use the new `UpdateTime` on `Buff`.